include/rocksdb/listener.h

```diff
@@ -348,6 +355,18 @@ class EventListener {
   virtual void OnExternalFileIngested(
       DB* /*db*/, const ExternalFileIngestionInfo& /*info*/) {}
 
+  // A call-back function for RocksDB which will be called before setting the
+  // background error status to a non-OK value. The new background error status
+  // is provided in `bg_error` and can be modified by the callback. E.g., a
+  // callback can suppress errors by resetting it to Status::OK(), thus
+  // preventing the database from entering read-only mode.
```

Comment:
Also explain that, if users reset it to OK, there isn't a guarantee following flush and compaction will be scheduled and executed as expected.